code/modules/mob/living/carbon/carbon.dm

```diff
@@ -484,55 +486,53 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 
 
 	if(!vent_found)
-		for(var/obj/machinery/atmospherics/machine in range(1,src))
+		for(var/obj/machinery/atmospherics/machine in range(1, src))
 			if(is_type_in_list(machine, GLOB.ventcrawl_machinery) && machine.can_crawl_through())
 				vent_found = machine
 				break
 
-	if(vent_found)
-		if(vent_found.parent && (vent_found.parent.members.len || vent_found.parent.other_atmosmch))
-			visible_message("<span class='notice'>[src] begins climbing into the ventilation system...</span>", \
-							"<span class='notice'>You begin climbing into the ventilation system...</span>")
+	if(!vent_found)
+		to_chat(src, "<span class='warning'>This ventilation duct is not connected to anything!</span>")
+		return
 
-			if(!do_after(src, 45, target = src))
-				return
+	if(vent_found.parent && (length(vent_found.parent.members) || vent_found.parent.other_atmosmch))
```

Comment:
Any reason not to also guard statement this check?

Reply:
Made it a guard clause